test/spec/spec.filerotatetransport.js

```diff
@@ -0,0 +1,316 @@
+            clock = sinon.useFakeTimers(Date.parse(new Date(2016, 10, 25, 12, 11, 10)));
+            transport = new FileRotateTransport({ filename: '/path/test.log', dateRotate: true });
+            sinon.stub(fs, 'access').yields({ message: 'file not found' });
+            sinon.stub(fs, 'rename').yields();
+            sinon.stub(FileRotateTransport.prototype, '_dateRotateUpdateDay');
+            cb = sinon.stub();
+        });
+
+        afterEach(function () {
+            clock.restore();
+            fs.access.restore();
+            fs.rename.restore();
+            FileRotateTransport.prototype._dateRotateUpdateDay.restore();
+        });
+
+        it('calls _dateRotateUpdateDay', function () {
+            transport._dateRotateLog(cb);
+            FileRotateTransport.prototype._dateRotateUpdateDay.should.have.been.calledOnce;
+        });
+
+        it('calls access to check if the archive log already exists', function () {
+            transport._dateRotateLog(cb);
+            fs.access.should.have.been.calledWithExactly('/path/test-2016-11-25.log', sinon.match.func);
+        });
+
+        it('calls callback if log file already exists', function () {
+            fs.access.yields();
+            transport._dateRotateLog(cb);
+            fs.rename.should.not.have.been.called;
+            cb.should.have.been.calledWithExactly({message: 'Archive already exists'});
+        });
+
+        it('renames the log file to an archive name', function () {
+            transport._dateRotateLog(cb);
+            fs.rename.should.have.been.calledWithExactly('/path/test.log', '/path/test-2016-11-25.log', sinon.match.func);
+        });
+
+        it('calls the callback with return from rename', function () {
+            fs.rename.yields('value');
+            transport._dateRotateLog(cb);
+            cb.should.have.been.calledWithExactly('value');
+        });
+    });
+
+    describe('_dateRotateGlob', function () {
+        it('should be the glob module', function () {
+            FileRotateTransport.prototype._dateRotateGlob.should.equal(glob);
+        });
+    });
+
+    describe('_dateRotateRemoveOldFiles', function () {
+        var transport, cb;
+
+        beforeEach(function () {
+            transport = new FileRotateTransport({ filename: '/path/test.log', dateRotate: true });
+            sinon.stub(async, 'each');
+            sinon.stub(FileRotateTransport.prototype, '_dateRotateGlob');
+            cb = sinon.stub();
+        });
+
+        afterEach(function () {
+            async.each.restore();
+            FileRotateTransport.prototype._dateRotateGlob.restore();
+        });
+
+        it('calls the callback without listing files if maxFiles is zero', function () {
+            transport.maxFiles = 0;
+            transport._dateRotateRemoveOldFiles(cb);
+            cb.should.have.been.calledWithExactly();
+            FileRotateTransport.prototype._dateRotateGlob.should.not.have.been.called;
```

Comment:
when maxFiles is zero old files are not deleted, all rotated logs are left on disk, i.e. glob async isn't called to list files to delete.